src/redux/books/booksSlice.js

```diff
@@ -0,0 +1,41 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  books: [
+    {
+      item_id: 'item1',
+      title: 'The Great Gatsby',
+      author: 'John Smith',
+      category: 'Fiction',
+    },
+    {
+      item_id: 'item2',
+      title: 'Anna Karenina',
+      author: 'Leo Tolstoy',
+      category: 'Fiction',
+    },
+    {
+      item_id: 'item3',
+      title: 'The Selfish Gene',
+      author: 'Richard Dawkins',
+      category: 'Nonfiction',
+    },
+  ],
+};
+const bookSlice = createSlice({
+  name: 'books',
+  initialState,
+  reducers: {
+    addBook: (state, action) => {
+      state.books.push(action.payload);
+    },
```

Comment:
- [OPTIONAL] I noticed you're using the `push` method to add elements to an array. Consider using the `concat` method instead. Unlike `push`, `concat` returns a new array with the added element, preserving the original array. This can prevent unintended side effects and make your code more predictable. Keep coding awesomely! 👍📚

Reply:
Thank you. This is very helpful